packages/cli/src/cmds/preview/preview.ts

```diff
@@ -681,7 +707,7 @@ export default class Preview extends YargsCommand {
         backendWatchTask.stopCb,
         this.serviceLogWriter
       ),
-      await botStartTask?.task.waitFor(
+      botStartTask?.task.waitFor(
         constants.botStartPattern,
```

Comment:
For `teamsfx` script, it's possible the start pattern is changed. How about adding a default timeout (e.g., 30s) for those `teamsfx` tasks in case start pattern is changed?

Reply:
Good point. Added timeout for `waitFor` and only set 30s for bot start task.